### PR TITLE
Fix return and param type

### PR DESCRIPTION
**Pull request by Lainow**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 

When I tried to update some plugins to phpstan 5, I got errors related to incorrect return types and parameters.

### CODE REVIEW

File: src/DropdownTranslation.php [hunk 1]
```diff
@@ -201,10 +201,10 @@ public function post_addItem()
     /**
      * Return the number of translations for a field in a language
      *
-     * @param itemtype
-     * @param items_id
-     * @param field
-     * @param language
+     * @param $itemtype
```
Comment (trasher) [inline src/DropdownTranslation.php:204]: Parameter type must be declared

Comment (trasher) [reply src/DropdownTranslation.php:204]: (remark is for all the fix of the same ink in the PR)

**Review by trasher — CHANGES_REQUESTED** [review]
See my comments

File: src/DropdownTranslation.php [hunk 2]
```diff
@@ -201,10 +201,10 @@ public function post_addItem()
     /**
      * Return the number of translations for a field in a language
      *
-     * @param $itemtype
-     * @param $items_id
-     * @param $field
-     * @param $language
+     * @param string $itemtype
+     * @param int|string $items_id
```
Comment (cedric-anne) [inline src/DropdownTranslation.php:205]:
```suggestion
     * @param int $items_id
```

File: src/DropdownTranslation.php [hunk 3]
```diff
@@ -247,8 +247,8 @@ public static function getNumberOfTranslationsForItem($item)
     /**
      * Check if a field's translation can be added or updated
      *
-     * @param $input          translation's fields
-     * @param $add    boolean true if a transaltion must be added, false if updated (true by default)
+     * @param mixed $input          translation's fields
```
Comment (cedric-anne) [inline src/DropdownTranslation.php:250]:
```suggestion
     * @param bool $input          translation's fields
```

File: src/DropdownTranslation.php [hunk 4]
```diff
@@ -270,8 +270,8 @@ public function checkBeforeAddorUpdate($input, $add = true)
     /**
      * Generate completename associated with a tree dropdown
      *
-     * @param $input array    of user values
-     * @param $add   boolean  true if translation is added, false if update (tgrue by default)
+     * @param mixed $input array    of user values
```
Comment (cedric-anne) [inline src/DropdownTranslation.php:273]:
```suggestion
     * @param array $input     array of user values
```

**Review by trasher — APPROVED** [review]

File: src/Fieldblacklist.php [hunk 5]
```diff
@@ -398,10 +398,10 @@ public function selectValues($field = '')
     /**
      * Check if a field & value are blacklisted or not
      *
-     * @param $itemtype      itemtype of the blacklisted field
-     * @param $entities_id   the entity in which the field must be saved
-     * @param $field         the field to check
-     * @param $value         the field's value
+     * @param string $itemtype      itemtype of the blacklisted field
+     * @param int|string $entities_id   the entity in which the field must be saved
```
Comment (cedric-anne) [inline src/Fieldblacklist.php:402]:
```suggestion
     * @param int $entities_id   the entity in which the field must be saved
```

File: src/Fieldblacklist.php [hunk 6]
```diff
@@ -398,10 +398,10 @@ public function selectValues($field = '')
...
+     * @param string $field         the field to check
+     * @param array|string $value         the field's value
```
Comment (cedric-anne) [inline src/Fieldblacklist.php:404]:
```suggestion
     * @param string $value         the field's value
```

File: src/ReminderTranslation.php [hunk 7]
```diff
@@ -310,7 +310,7 @@ public static function isReminderTranslationActive()
      * It be translated if translation if globally on and item is an instance of CommonDropdown
      * or CommonTreeDropdown and if translation is enabled for this class
      *
-     * @param $item the item to check
+     * @param CommonDBTM $item the item to check
```
Comment (cedric-anne) [inline src/ReminderTranslation.php:313]:
```suggestion
     * @param CommonGLPI $item the item to check
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**